### PR TITLE
Mobile loading

### PR DESCRIPTION
**Pull request by MarcoPolo**
@keybase/react-hackers 

Adds loading screen for user bio and proofs and a component that handles transistions.

Preview 1/3 speed to demo animation
profile loading

### CODE REVIEW

File: shared/profile/render.native.js [hunk 1]
```diff
@@ -121,51 +123,82 @@ class Render extends Component<void, Props, State> {
...
-            onAcceptProofs={this.props.onAcceptProofs}
-          />
+          <Box style={{...globalStyles.flexBoxColumn, position: 'relative'}}>
+            <Box style={{position: 'absolute', backgroundColor: trackerStateColors.header.background, top: 0, left: 0, right: 0, height: 56}} />
+            <LoadingWrapper
+              style={{minHeight: 220}}
+              duration={500}
+              loading={this.props.loading}
+              loadingComponent={(
+                <UserBio
```
Comment (chrisnojima) [inline shared/profile/render.native.js]: does this really need to be duplicated? can't doneLoadingComponent = loadingComponent if its not specified or something?

File: shared/common-adapters/avatar.native.js [hunk 2]
```diff
@@ -10,31 +10,36 @@ import {iconMeta} from './icon.constants'
...
   }
 
   render () {
     const {size} = this.props
-    const uri = {uri: shared.createAvatarUrl(this.props)}
+    const uri = shared.createAvatarUrl(this.props)
     const propsOpacity = this.props.hasOwnProperty('opacity') ? this.props.opacity : 1.0
     const opacity = this.state.avatarLoaded ? propsOpacity : 0
 
+    const showNoAvatar = (!this.props.loadingColor && (!this.state.avatarLoaded || !uri)) ||
```
Comment (chromakode) [inline shared/common-adapters/avatar.native.js]: This boolean expression is a bit hard to parse. Can it be simplified, or broken down into named subexpressions?

File: shared/common-adapters/loading-wrapper.native.js [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+}
+
+class LoadingWrapper extends Component<void, Props, State> {
+  state: State;
+
+  constructor (props: Props) {
+    super(props)
+    this.state = {
+      opacity: new Animated.Value(1),
+      loadingActive: this.props.loading,
```
Comment (chromakode) [inline shared/common-adapters/loading-wrapper.native.js]: Is it necessary to default this state value in this way? Looks like it gets set to true when the `loading` prop transitions values anyway?

File: shared/common-adapters/loading-wrapper.native.js [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+  }
+
+  _doAnimation () {
+    const {duration} = this.props
+    this.setState({loadingActive: true})
+    this.props.clearTimeout(this.state.loadingActiveTimeoutId)
+
+    this.state.opacity.setValue(1)
+    Animated.parallel(
+      [[this.state.opacity, 0]].map(
```
Comment (chromakode) [inline shared/common-adapters/loading-wrapper.native.js]: Trying to understand why `map` is used here -- is the added complexity desired for some reason?

File: shared/common-adapters/loading-wrapper.native.js [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+        ([a, toValue]) => Animated.timing(a, {duration, toValue})
+      )
+    ).start()
+
+    const loadingActiveTimeoutId = this.props.setTimeout(() => this.setState({loadingActive: false}), this.props.duration)
```
Comment (chromakode) [inline shared/common-adapters/loading-wrapper.native.js]: Is there a completion callback in `Animated` we could hook into instead of using a timeout?

File: shared/styles/style-guide.native.js [hunk 6]
```diff
@@ -90,24 +100,30 @@ const util = {
   textDecoration: (type: string) => ({
     textDecorationLine: type,
   }),
+  loadingTextStyle: {
+    backgroundColor: globalColors.lightGrey,
+    // borderColor: 'red',
```
Comment (chromakode) [inline shared/styles/style-guide.native.js]: Was this intended to be removed?

File: shared/common-adapters/loading-wrapper.native.js (hunk 4 above)
Comment (MarcoPolo) [inline shared/common-adapters/loading-wrapper.native.js]: It was to do multiple animations (fadeIn while fading out) but I decided against it.

We can make this just the timing code below

File: shared/common-adapters/loading-wrapper.native.js (hunk 3 above)
Comment (MarcoPolo) [inline shared/common-adapters/loading-wrapper.native.js]: You could have loading initially be false.

Not sure if that answers your question.

File: shared/common-adapters/loading-wrapper.native.js (hunk 5 above)
Comment (MarcoPolo) [inline shared/common-adapters/loading-wrapper.native.js]: Ooo there is!

File: shared/styles/style-guide.native.js (hunk 6 above)
Comment (MarcoPolo) [inline shared/styles/style-guide.native.js]: yes...

File: shared/common-adapters/avatar.native.js (hunk 2 above)
Comment (MarcoPolo) [inline shared/common-adapters/avatar.native.js]: Is this better?

```
const showLoadingColor = (this.props.loadingColor && !this.state.avatarLoaded) || this.props.forceLoading
const showNoAvatar = (!showLoadingColor && (!this.state.avatarLoaded || !uri || this.state.errored))
```

Comment (chromakode) [inline shared/common-adapters/avatar.native.js]: Yes!

File: shared/common-adapters/loading-wrapper.native.js (hunk 3 above)
Comment (chromakode) [inline shared/common-adapters/loading-wrapper.native.js]: Ah I see, this is to determine whether the loading state is initially displayed. I retract my comment!

File: shared/common-adapters/avatar.native.js [hunk 7]
```diff
@@ -36,9 +36,8 @@ class Avatar extends Component<void, Props, State> {
     const propsOpacity = this.props.hasOwnProperty('opacity') ? this.props.opacity : 1.0
     const opacity = this.state.avatarLoaded ? propsOpacity : 0
 
-    const showNoAvatar = (!this.props.loadingColor && (!this.state.avatarLoaded || !uri)) ||
-      (this.state.avatarLoaded && this.state.errored)
     const showLoadingColor = (this.props.loadingColor && !this.state.avatarLoaded) || this.props.forceLoading
+    const showNoAvatar = (!showLoadingColor && (!this.state.avatarLoaded || !uri || this.state.errored))
```
Comment (chromakode) [inline shared/common-adapters/avatar.native.js]: :sheep: Don't need the outer parens

File: shared/common-adapters/loading-wrapper.native.js [hunk 8]
```diff
@@ -2,7 +2,6 @@
 import React, {Component} from 'react'
 import {Animated} from 'react-native'
 import Box from './box'
-import Button from './button'
 import HOCTimers from './hoc-timers'
```
Comment (chromakode) [inline shared/common-adapters/loading-wrapper.native.js]: Is this still required?

Comment (MarcoPolo) [inline shared/common-adapters/loading-wrapper.native.js]: nope!

File: shared/common-adapters/avatar.native.js (hunk 7 above)
Comment (MarcoPolo) [inline shared/common-adapters/avatar.native.js]: My lisp is showing! 😮

**Comment by chromakode** [comment]
:+1:
